presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -416,12 +447,54 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     failureDetector,
                     nodeTaskMap,
                     stageSchedulers,
-                    stageLinkages);
+                    stageLinkages,
+                    stagedDependencies,
+                    schedulingQueue);
             stages.addAll(subTree);
 
             SqlStageExecution childStage = subTree.get(0);
             childStagesBuilder.add(childStage);
         }
+
+        for (Reference<StageExecutionPlan> dependency : plan.getDependencies()) {
+            PlanFragmentId fragmentId = dependency.get().getFragment().getId();
+            if (stagedDependencies.contains(fragmentId)) {
+                continue;
+            }
+
+            StageExecutionPlan subStagePlan = dependency.get();
+            List<SqlStageExecution> subTree = createStages(
+                    // dependent stage doesn't produce any output
+                    ((fragment, tasks, noMoreExchangeLocations) -> {}),
+                    nextStageId,
+                    locationFactory,
+                    subStagePlan.withBucketToPartition(bucketToPartition),
```

Comment:
This looks off. Since `bucketToPartition` are shared between stage and its direct sub-stage when joining a bucketed table with a unbucketed table.